### PR TITLE
Add unit tests for org.springframework.boot.configurationprocessor.json

### PR DESCRIPTION
**Pull request by ThomasPerkins1123**
Hi,

I've analysed your code base and noticed that `org.springframework.boot.configurationprocessor.json.JSONTokener` in the `spring-boot-configuration-processor` module is not fully tested.

I've written some tests that cover this class with the help of [Diffblue Cover](https://www.diffblue.com/opensource).

Hopefully, these tests should help you detect any regressions caused by future code changes. If you would find it useful to have additional tests written for this repository, I would be more than happy to look at other particular classes that you consider important.

### CODE REVIEW

**Comment by paulkennethkent** [comment]
@ThomasPerkins1123 I might be a bit slow as it's a Monday but I can't find the associated class file for these tests?

**Comment by petersear** [comment]
I also had trouble finding the source file. Is this it?

` spring-boot/spring-boot-project/spring-boot-tools/spring-boot-configuration-processor/src/json-shade/java/org/springframework/boot/configurationprocessor/json/JSONTokener.java`

(in which case it should be relocated in the test directory structure)

**Comment by ThomasPerkins1123** [comment]
@petersear Yes that is the class being tested. The tests where put here as this is where the eclipse plugin put the tests but the class itself is in a bit of a weird location so I wasn't sure of a better location to put them.

**Comment by petersear** [comment]
That does look like the best area to put the tests to me, having double-checked. Apologies for the confusion.

File: spring-boot-project/spring-boot-tools/spring-boot-configuration-processor/src/test/java/org/springframework/boot/configurationprocessor/json/JSONTokenerTest.java [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+
+	@Test
+	public void testNextCleanInternal7() throws JSONException {
+		JSONTokener jsonTokener = new JSONTokener("/*foo");
+		thrown.expect(JSONException.class);
+		Assert.assertEquals(0, jsonTokener.nextClean());
+	}
+
+	@Test
+	public void testSkipTo1() {
```
Comment (petersear) [inline spring-boot-project/spring-boot-tools/spring-boot-configuration-processor/src/test/java/org/springframework/boot/configurationprocessor/json/JSONTokenerTest.java:246]: Maybe assert a non-zero expected value.

**Review by paulkennethkent — APPROVED** [review]

**Review by petersear — APPROVED** [review]

**Pull request closed without merging**